llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -824,7 +824,15 @@ namespace llvm {
 Value *createStepForVF(IRBuilderBase &B, Type *Ty, ElementCount VF,
                        int64_t Step) {
   assert(Ty->isIntegerTy() && "Expected an integer step");
-  return B.CreateElementCount(Ty, VF.multiplyCoefficientBy(Step));
+  ElementCount VFxStep = VF.multiplyCoefficientBy(Step);
+  assert(isPowerOf2_64(VF.getKnownMinValue()) && "must pass power-of-2 VF");
+  if (VF.isScalable() && isPowerOf2_64(Step)) {
+    return B.CreateShl(
+        B.CreateVScale(Ty),
+        ConstantInt::get(Ty, Log2_64((VF * Step).getKnownMinValue())), "",
```

Comment:
Looks like it's not reusing `VFxStep` - I think it got lost from the previous commit.

Reply:
Ah yes, added back, thanks!